drivers/bluetooth/hci/rpmsg.c

```diff
@@ -45,6 +45,17 @@ static bool is_hci_event_discardable(const uint8_t *evt_data)
 		switch (subevt_type) {
 		case BT_HCI_EVT_LE_ADVERTISING_REPORT:
 			return true;
+#if defined(CONFIG_BT_EXT_ADV)
+		case BT_HCI_EVT_LE_EXT_ADVERTISING_REPORT:
+		{
+			const struct bt_hci_evt_le_ext_advertising_report *ext_adv =
+				(void *)&hci_buf[3];
+
+			return (ext_adv->num_reports == 1) &&
+				   ((ext_adv->adv_info->evt_type &
+					 BT_HCI_LE_ADV_EVT_TYPE_LEGACY) != 0);
```

Comment:
@cvinayak does the Zephyr LL also only support a single report per event? 

Should this be something like this to future proof the check? 
```suggestion
			for (uint8_t i = 0U; i < ext_adv->num_reports; i++) {
				if (ext_adv->adv_info[i].evt_type & BT_HCI_LE_ADV_EVT_TYPE_LEGACY == 0) {
					/* Non-legacy adv reports cannot be discarded */
					return false;
				}
			}

			return true;
```

Reply:
I don't fully grasp all the different HCI implemenations, but wouldn't this very awkward in the h4 implementation?  because there you would have to wait for many bytes to come in over uart before you could determine if this event is discardable..

but yeah, Im also currious to if ZLL ever gives reports with num_reports>1

Reply:
Good question. This PR seems to also only add this for the `rpmsg` HCI driver. What about the others?

Reply:
> @cvinayak does the Zephyr LL also only support a single report per event?
> 
> Should this be something like this to future proof the check?

Zephyr LL only supports preparing single report per HCI event. 

A simple `for` loop cannot be used, as the event is length encoded using `struct bt_hci_evt_le_advertising_info` as header containing the `length` of each `data`.

Reply:
> > @cvinayak does the Zephyr LL also only support a single report per event?
> > Should this be something like this to future proof the check?
> 
> Zephyr LL only supports preparing single report per HCI event.
> 
> A simple `for` loop cannot be used, as the event is length encoded using `struct bt_hci_evt_le_advertising_info` as header containing the `length` of each `data`.

That's a good point, and looping over it as it is would require significant more effort.

Reply:
Host does the parsing already for a HCI event with multiple reports (for controllers that support), and as it needs the non-legacy reports to combine the fragments, host could drop the legacy report when parsing. But, that beats the purpose of discard-able HCI event if the whole event has to be passed to host from the HCI driver.